media_player/alexa.py

```diff
@@ -471,15 +493,28 @@ def device_state_attributes(self):
 
 
 class AlexaLogin():
-    def __init__(self, url, email, password):
+    def __init__(self, url, email, password, hass):
         self._url = url
         self._email = email
         self._password = password
         self._session = None
         self._data = None
         self.status = None
-
-        self.login()
+        self._cookiefile = hass.config.path("{}.pickle".format(ALEXA_DATA))
+        cookies = None
+        if exists(self._cookiefile):
```

Comment:
hass.config.path already returns None if the file doesn't exist. So no need for the os.path exists() function here. 

You can simply do 

```
self._cookiefile = hass.config.path("{}.pickle".format(ALEXA_DATA))
cookies = None
if self._cookiefile:
```